infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -131,10 +187,21 @@ def restore_database(backup_fpath):
     )
     run_shell_command(create_db_cmd, env_vars=pg_env_vars)
 
+    # find the most recent backup or verify the specify backup exists
+    print("Getting backup file from S3")
+    session = create_s3_session()
+    if backup_obj_name is None:
+        backup_obj_name = get_most_recent_backup_obj_name(session)
+    elif backup_obj_name not in list_bucket_objects(session, BACKUP_BUCKET):
+        return False
```

Comment:
Is this return value being used anywhere? I'm assuming that if we get to this point, the whole restore process failed. This is actually a great use case for raising an exception. It's typically bad practice to raise exceptions from the standard library. So, somewhere towards the top of the file, I'd make an exception class like this:

```python
class DatabaseBackupError(Exception):
    """Raised when an error occurs in the database backup or restore process"""
    ...
```

Then, here, you can do:

```python
elif backup_obj_name not in list_bucket_objects(session, BACKUP_BUCKET):
    raise DatabaseBackupError(f"No database backup was found in the {BUCKET_NAME} bucket with object name {backup_obj_name}")
```

If we want to react to this error, we can catch it somewhere else, but I can't think of what we'd do if we caught it at this point. I think we'd actually want this to crash the backup container to inform us that something is wrong with our configuration.

Reply:
I love that `list_bucket_objects()` function. Do you happen to know if it struggles with "page size limits"? Like if there are `10,000` objects in the bucket, will the function really return all `10,000`? Honestly, if we have that many backups stored that's a problem, so even if it does struggle that's fine. It may break, though if it can only return `10`, for example.

Reply:
^^^ answered.

Reply:
Instead of having to build our own exception, I figured we could just pass it in anyways and then if the object is not in the bucket boto3 will throw the error for us. I am happy to create our own if youd rather we did that

Reply:
I changed my mind and added a custom exception